### PR TITLE
Dependency and workflow updates for Palm

### PR DESCRIPTION
**Pull request by aphilipharrell**
Frontend updates related to Palm upgrade:
- Update dependencies
- Update React Router redirects & links for new version
- Update PropTypes defs for objects and arrays
- Refactor hooks to appease the linter (no hooks in conditionals)

### CODE REVIEW

File: src/index.jsx [hunk 1]
```diff
@@ -58,4 +65,16 @@ initialize({
...
+  handlers: {
+    config: () => (
+      mergeConfig({
+        SITE_NAME: 'MOGC Partners',
+        TERMS_OF_SERVICE_URL: `${getConfig().LMS_BASE_URL}/tos`,
+        PRIVACY_POLICY_URL: `${getConfig().LMS_BASE_URL}/privacy`,
+        SUPPORT_EMAIL: 'fix@me.com',
+        SUPPORT_URL: 'https://teamdynamix.umich.edu/TDClient/187/Portal/Requests/TicketRequests/NewForm?ID=T0l23Fz2XA4_&RequestorType=Service',
+        ENABLE_ACCESSIBILITY_PAGE: 'https://teamdynamix.umich.edu/TDClient/187/Portal/KB/ArticleDet?ID=10934',
+      })
```
Comment (aphilipharrell) [inline src/index.jsx:77]: Need to confirm `SITE_NAME` AND `SUPPORT_EMAIL`.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -24,7 +24,5 @@ jobs:
       run: npm run build
     - name: i18n_extract
       run: npm run i18n_extract
-    - name: is-es5
-      run: npm run is-es5
```
Comment (aphilipharrell) [inline .github/workflows/ci.yml:28]: This is no longer included in `frontend-template-application` ([here](https://github.com/openedx/frontend-template-application/blob/master/package.json)) and I don't think we need it, but LMK if that seems wrong.

File: webpack.dev-tutor.config.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  output: {
+    hashFunction: 'xxhash64',
+  },
+};
```
Comment (aphilipharrell) [inline webpack.dev-tutor.config.js:5]: Fixes an issue with the build process: https://stackoverflow.com/a/69394785

File: package.json [hunk 4]
```diff
@@ -14,7 +14,6 @@
     "build": "fedx-scripts webpack",
     "prepare": "husky install",
     "i18n_extract": "BABEL_ENV=i18n fedx-scripts babel src --quiet > /dev/null",
-    "is-es5": "es-check es5 ./dist/*.js",
```
Comment (aphilipharrell) [inline package.json:17]: Perhaps is still worthwhile to include here?

File: .env.development [hunk 5]
```diff
@@ -19,3 +19,5 @@ REFRESH_ACCESS_TOKEN_ENDPOINT='http://localhost:18000/login_refresh'
 SEGMENT_KEY=''
 SITE_NAME=localhost
 USER_INFO_COOKIE_NAME='edx-user-info'
+APP_ID=
+MFE_CONFIG_API_URL=
```
Comment (aphilipharrell) [inline .env.development:23]: This is what edX contributors added to the dev envs in related runtime config PRs. The frontend app still calls `http://apps.local.overhang.io:1998/api/mfe_config/v1?mfe=mogc-partners` which is proxied to `http://local.overhang.io:8000`.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (michaelwheeler) [reply .github/workflows/ci.yml:28]: Seems okay to me.

File: package.json (hunk 4 above)
Comment (michaelwheeler) [reply package.json:17]: Nah. Looks like [it was removed from the template ](https://github.com/openedx/frontend-template-application/commit/51a147fc872231f80b8f3762754aff91b2773eaf) at the same time the CI job was.

File: initialize.js
Comment (michaelwheeler) [inline initialize.js:1]: What is this empty file for?

File: src/common/ResponsiveBreadcrumb.jsx [hunk 6]
```diff
@@ -1,27 +1,27 @@
...
-  // See: https://paragon-openedx.netlify.app/components/breadcrumb/#with-custom-link-element
   const newLinks = links.map(
-    link => ({ label: link.label, url: addBasepath(link.url) }),
+    link => ({ label: link.label, to: link.url }),
   );
   return (
     <Breadcrumb
       {...props}
       links={newLinks}
+      linkAs={Link}
```
Comment (michaelwheeler) [inline src/common/ResponsiveBreadcrumb.jsx:16]: Thanks for updating this!

**Review by michaelwheeler — CHANGES_REQUESTED** [review]
Thanks @aphilipharrell. Just one question to resolve.

I also noticed that there were a lot of formatting changes. Were those required to pass checks with updated `frontend-build` and `frontend-platform`?

File: initialize.js
Comment (aphilipharrell) [reply initialize.js:1]: Nuthin', will delete - left in by accident, thank you!

**Comment by aphilipharrell** [comment]
@michaelwheeler thanks for reviewing! The formatting changes weren't required but I did run Prettier locally to auto-format some of the other changes. Happy to undo them if you think they made things worse.

**Comment by michaelwheeler** [comment]
> @michaelwheeler thanks for reviewing! The formatting changes weren't required but I did run Prettier locally to auto-format some of the other changes. Happy to undo them if you think they made things worse.

No need to undo, and in fact I would be psyched if we could add Prettier to our CI someday. When possible in the future though I think I'd prefer to keep formatting-only changes in their own PR.

**Review by michaelwheeler — APPROVED** [review]

File: package.json [hunk 7]
```diff
@@ -31,36 +30,39 @@
...
+    "@edx/frontend-component-footer": "13.1.0",
+    "@edx/frontend-component-header": "^5.0.2",
+    "@edx/frontend-platform": "^8.0.0",
+    "@fortawesome/fontawesome-svg-core": "6.5.1",
+    "@fortawesome/free-brands-svg-icons": "6.5.1",
+    "@fortawesome/free-regular-svg-icons": "6.5.1",
+    "@fortawesome/free-solid-svg-icons": "6.5.1",
+    "@fortawesome/react-fontawesome": "0.2.0",
+    "@openedx/brand-openedx": "^1.2.3",
+    "@openedx/frontend-build": "^14.0.3",
```
Comment (aphilipharrell) [inline package.json:43]: This dependency update will also need to be reverted - although the [package has officially moved](https://www.npmjs.com/package/@edx/frontend-build), the Tutor MFE plugin hasn't been updated to use it, and still uses the [edx package](https://github.com/overhangio/tutor-mfe/blob/15a0c6b2952361b59679dd60479d54a1ce689d2d/tutormfe/templates/mfe/apps/mfe/webpack.dev-tutor.config.js#L7), which causes the MFE container to fail to start in dev.

**Pull request merged**